src/app_service/service/community/service.nim

```diff
@@ -484,42 +487,30 @@ QtObject:
         self.events.emit(SIGNAL_COMMUNITY_MEMBERS_CHANGED, 
         CommunityMembersArgs(communityId: community.id, members: community.members))
 
-      self.allCommunities[community.id] = community
+      self.communities[community.id] = community
       self.events.emit(SIGNAL_COMMUNITIES_UPDATE, CommunitiesArgs(communities: @[community]))
 
-      if(not self.joinedCommunities.hasKey(community.id)):
-        if (community.joined and community.isMember):
-          self.joinedCommunities[community.id] = community
-          self.events.emit(SIGNAL_COMMUNITY_JOINED, CommunityArgs(community: community, fromUserAction: false))
```

Comment:
shouldn't we send this signal too?
Won't it break the logic, where we wait for a SIGNAL_COMMUNITY_JOINED signal?

Reply:
Good point. I thought the condition above `if not self.communities.hasKey(community.id)` would be good enough to call `SIGNAL_COMMUNITY_JOINED`, but I forgot that it's possible to already know the community ad then get a signal that we joined it. I added the signal back at the bottom.